src/Adapter/Presenter/Product/ProductLazyArray.php

```diff
@@ -776,7 +776,7 @@ public function addQuantityInformation(
                     : Configuration::get('PS_LABEL_OOS_PRODUCTS_BOA', $language->id);
                 $this->product['availability_date'] = $product['available_date'];
                 $this->product['availability'] = 'available';
-            } elseif ($product['quantity_wanted'] > 0 && $product['quantity'] > 0) {
+            } elseif ($product['quantity_wanted'] > 1 && $product['quantity'] > 0) {
```

Comment:
Should be for a better understanding. 
```suggestion
            } elseif ($product['quantity'] - $product['quantity_wanted'] < 0) {
```